picky-asn1-x509/src/pkcs12/safe_bag.rs

```diff
@@ -0,0 +1,455 @@
+mod certificate;
+mod crl;
+mod key;
+mod secret;
+
+use crate::oids;
+use crate::pkcs12::{Pkcs12Attribute, SafeContents};
+use core::fmt;
+use oid::ObjectIdentifier;
+use picky_asn1::wrapper::{Asn1SetOf, ExplicitContextTag0, ObjectIdentifierAsn1};
+use picky_asn1_der::Asn1RawDer;
+use serde::{de, ser};
+
+pub use certificate::*;
+pub use crl::*;
+pub use key::*;
+pub use secret::*;
+
+/// [PKCS #12: Personal Information Exchange Syntax Standard](https://datatracker.ietf.org/doc/html/rfc7292#section-4.2)
+///
+/// SafeBag is the atomic node of PKCS#12 structure, it could contain concrete data such as
```

Comment:
Nit: it’s not exactly clear to me what atomic means in this situation. Is my understanding correct?
```suggestion
/// SafeBag is the smallest unit of information in PKCS#12 SafeContents, it could contain concrete data such as
```
Though I’m not sure we can say it’s the smallest unit of information since a `SafeBag` can contain other `SafeBag`s recursively.
```suggestion
/// SafeBag a building block of PKCS#12 structure, it could contain concrete data such as
```

Reply:
Yeah, `a building block` sounds right